### PR TITLE
Add OpenReview scraper query method

### PR DESCRIPTION
**Pull request by satyaog**

### CODE REVIEW

File: paperoni/sources/scrapers/openreview.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+        title: Option = [],
+        # Maximal number of results per query
+        block_size: Option & int = 100,
+        # Maximal number of results to return
+        limit: Option & int = 10000,
+    ):
+        title = " ".join(title)
+
+        if author and title:
+            raise QueryError("Cannot query both author and title")
```
Comment (breuleux) [inline paperoni/sources/scrapers/openreview.py:37]: That was a limitation for Semantic Scholar, but is it a limitation for OpenReview?

File: paperoni/sources/scrapers/openreview.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+        client = openreview.Client(baseurl='https://api.openreview.net')
+        params = {
+            "content": {"venueid": 'NeurIPS.cc/2021/Conference'},
```
Comment (breuleux) [inline paperoni/sources/scrapers/openreview.py:41]: Fill with `--venue` parameter.

File: paperoni/sources/scrapers/openreview.py [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+        # Author to query
+        # [alias: -a]
+        # [nargs: +]
+        author: Option = [],
+        # Title of the paper
+        # [alias: -t]
+        # [nargs: +]
+        title: Option = [],
+        # Maximal number of results per query
+        block_size: Option & int = 100,
```
Comment (breuleux) [inline paperoni/sources/scrapers/openreview.py:30]: Does not appear to be used other than to override limit.

File: pyproject.toml [hunk 4]
```diff
@@ -20,6 +20,7 @@ pydantic = "^1.9.1"
 SQLAlchemy = "^1.4.39"
 ovld = "^0.3.2"
 questionary = "^1.10.0"
+openreview-py = {git = "https://github.com/openreview/openreview-py.git"}
```
Comment (breuleux) [inline pyproject.toml:23]: Is there a reason why it is installed from git? `pip install openreview-py` seems to work well enough.

File: paperoni/sources/scrapers/openreview.py (hunk 1 above)
Comment (satyaog) [reply paperoni/sources/scrapers/openreview.py:37]: Probably not but will check

File: paperoni/sources/scrapers/openreview.py (hunk 2 above)
Comment (satyaog) [reply paperoni/sources/scrapers/openreview.py:41]: That was debug line but you do see a value in filtering by venue?

File: pyproject.toml (hunk 4 above)
Comment (satyaog) [reply pyproject.toml:23]: I just followed what they suggest in their docs (https://docs.openreview.net/getting-started/using-the-api/installing-and-instantiating-the-python-client) but will check if the pypi package looks up to date

Comment (satyaog) [reply pyproject.toml:23]: pypi package is not up to date with master but it is with the latest version tag which is probably better

Comment (satyaog) [inline pyproject.toml:23]:
```suggestion
openreview-py = "^1.5.2"
```

File: paperoni/sources/scrapers/openreview.py [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+    ):
+        title = " ".join(title)
+
+        if author and title:
+            raise QueryError("Cannot query both author and title")
+
+        client = openreview.Client(baseurl='https://api.openreview.net')
+        params = {
+            "content": {"venueid": 'NeurIPS.cc/2021/Conference'},
+            "limit": min(block_size or 10000, limit),
```
Comment (satyaog) [inline paperoni/sources/scrapers/openreview.py:42]:
```suggestion
            "limit": min(block_size or limit, limit),
```

File: paperoni/sources/scrapers/openreview.py (hunk 3 above)
Comment (satyaog) [reply paperoni/sources/scrapers/openreview.py:30]: True but that's enough to set the step size for each page. `limit` is still used to limit the total number of results but the `limit` in the openreview param dict is analog to `block_size`

File: paperoni/sources/scrapers/openreview.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+    @tooled
+    def prepare(self, researchers):
+        pass
+
+    @tooled
+    def venues(self):
+        client = openreview.Client(baseurl='https://api.openreview.net')
+        for venue in client.get_group(id='venues').members:
+            yield venue
```
Comment (satyaog) [inline paperoni/sources/scrapers/openreview.py:123]: not sure how I can make this available as `mpap venues openreview`

File: paperoni/sources/model.py [hunk 7]
```diff
@@ -23,6 +23,11 @@ class VenueType(str, Enum):
     case_report = "case_report"
     clinical_trial = "clinical_trial"
     unknown = "unknown"
+    challenge = "challenge"
+    forum = "forum"
+    track = "track"
+    tutorials = "tutorials"
+    seminar = "seminar"
```
Comment (satyaog) [inline paperoni/sources/model.py:30]: Are those new types making sense?

Comment (breuleux) [reply paperoni/sources/model.py:30]: Yes, sure.

**Pull request merged**